### PR TITLE
Update significance model field and logic from architect meeting

### PR DESCRIPTION
**Pull request by MariusArhaug**
fyi @geirst

### CODE REVIEW

File: linguistics/src/main/java/com/yahoo/language/significance/impl/DefaultSignificanceModelRegistry.java [hunk 1]
```diff
@@ -23,24 +24,45 @@
...
     @Inject
-    public DefaultSignificanceModelRegistry(SignificanceConfig cfg) { this(new Builder(cfg)); }
-    private DefaultSignificanceModelRegistry(Builder b) {
+    public DefaultSignificanceModelRegistry(SignificanceConfig cfg) {
         this.models = new EnumMap<>(Language.class);
-        b.models.forEach((language, path) -> {
-            models.put(language,
-                    uncheck(() -> new DefaultSignificanceModel(path)));
-        });
+        ObjectMapper objectMapper = new ObjectMapper();
```
Comment (bjorncs) [inline linguistics/src/main/java/com/yahoo/language/significance/impl/DefaultSignificanceModelRegistry.java:31]: Use the shared object mapper instance from `com.yahoo.json.Jackson.mapper()` instead.

File: linguistics/src/main/java/com/yahoo/language/significance/impl/DefaultSignificanceModelRegistry.java [hunk 2]
```diff
@@ -23,24 +24,45 @@
...
+        for (var model : models) {
+            try {
+                SignificanceModelFile file = objectMapper.readValue(model.toFile(), SignificanceModelFile.class);
+                for (var pair : file.languages().entrySet()) {
+                    addModel(
+                            Language.fromLanguageTag(pair.getKey()), new DefaultSignificanceModel(pair.getValue(), file.id()));
+                }
+            } catch (IOException e) {
+                throw new RuntimeException("Failed to load model from " + model, e);
+            }
```
Comment (bjorncs) [inline linguistics/src/main/java/com/yahoo/language/significance/impl/DefaultSignificanceModelRegistry.java:59]: This code snippet is the same as in the other constructor and can be moved to a separate method to reduce code duplication.

File: linguistics/src/main/java/com/yahoo/language/significance/impl/DefaultSignificanceModelRegistry.java (hunk 1 above)
Comment (bjorncs) [reply linguistics/src/main/java/com/yahoo/language/significance/impl/DefaultSignificanceModelRegistry.java:31]: Nvm, the `container-core` is not a dependency of `linguistics`.

File: linguistics/src/main/java/com/yahoo/language/significance/impl/DocumentFrequencyFile.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
+package com.yahoo.language.significance.impl;
```
[the remark this answers is not in the document]
Comment (MariusArhaug) [reply linguistics/src/main/java/com/yahoo/language/significance/impl/DocumentFrequencyFile.java:1]: Added @author tag aswell

**Review by bjorncs — APPROVED** [review]

**Pull request merged**